relay-quotas/src/global.rs

```diff
@@ -1,102 +1,82 @@
-use std::sync::{Mutex, PoisonError};
+use std::future::Future;
 
 use itertools::Itertools;
 use relay_base_schema::metrics::MetricNamespace;
 use relay_redis::{PooledClient, RedisError, RedisScripts};
 
-use crate::RedisQuota;
+use crate::redis::RedisQuota;
+use crate::RateLimitingError;
 
 /// Default percentage of the quota limit to reserve from Redis as a local cache.
 const DEFAULT_BUDGET_RATIO: f32 = 0.001;
 
+/// A trait that exposes methods to check global rate limits.
+pub trait GlobalLimiter {
+    /// Returns the [`RedisQuota`]s that should be rate limited.
+    fn check_global_rate_limits<'a>(
```

Comment:
I decided to hide the `OwnedRedisQuota` type from the trait to provide flexibility to the implementers, at the cost of some inefficiency, since the service implementation must convert data from `RedisQuota` to `OwnedRedisQuota`.